src/lib/operators/join_hash.hpp

```diff
@@ -11,6 +11,14 @@
 
 namespace opossum {
 
+struct JoinHashPerformanceData final {
```

Comment:
I fear that we will add specialized performance data to more and more operators then. This will be complicated and confusing.

Reply:
I think it is valuable to have more fine grained performance data for our perf-hungry operators both for taking performance samples to build a cost model and to do performance debugging.

But I can take these out of this PR if you want to.

(Complicated and Confusing? How so?)

Reply:
It's for sure valuable. But I envision a future scenario where 50% percent of our 30+ different operator implementations have additional performance data.

Reply:
mmh I wouldn't say that's a bad thing, but probably not part of this PR, so it'll go for now.